Improve the visualization R script

laughedelic opened the pull request:
This addresses one problem that I encountered in https://github.com/era7bio/tcrbsilvia/issues/5 (regarding having spaces in columns of the TSV) and some improvements recommended by @rtobes (less elements and amino acid sequences instead of IDs). I don't know if all of this applies equally to the MIODx usage of this project, so you tell me.

* change to using only top-25 to make the visualization less crowded
* use different columns: nucleotide sequence & amino acid sequence (so
  that the chart will show amino acid seq-s instead of the
  representative seq-s IDs)
* use `cut` to choose columns of interest, because `awk` splits by any
  whitespace and it breaks if any column contains spaces

Here's how it looks after the change:

![output](https://user-images.githubusercontent.com/766656/36862513-4f85dd96-1d87-11e8-89ad-e651bb5f5541.png)

agarciamontoro commented:
One thing I just remembered: I think @rtobes told me to add the frequencies to the leaves labels, and this removes them. Do we want them, Raquel?

rtobes commented:
MioDx is independent of this.

This was a quick solution to deliver in time some useful phylogenetic results of CDR3 regions.

To have also the frequency of each leave (CDR3) would be good.

laughedelic commented:
> One thing I just remembered: I think @rtobes told me to add the frequencies to the leaves labels, and this removes them.

@agarciamontoro I'm confused about it. Where were the frequencies before?

agarciamontoro commented:
Oops, you're right, this was not here. I probably modified the R script when I did the test trees for MIODx but not commited those changes as it was something temporal.

Anyway, the frequencies are in the 4th column of the TSV, so I think this would do the trick (I have not tested it):
```R
...
# Choosing Frequency, CDR3 sequence and AA-sequence columns
"cut -f 4,5,6",
# Output the information as FASTA with Frequency and Amino-acid seq-s as headers
"awk '{print \\">\\"$$1 - $$3\\"\\\\n\\"$$2;}'",
...
```

laughedelic commented:
Almost... :trollface: awk is tricky! It needs quotes for the dash `" - "` (plus escaping):

```
awk '{print \">\"$1\" - \"$3\"\\n\"$2;}'
```

The problem with this is that the labels are too long now:

![output](https://user-images.githubusercontent.com/766656/36910050-ef787af8-1e3f-11e8-8207-8af0e63183e4.png)

@rtobes what do you think?

rtobes commented:
Only 2 decimals

laughedelic commented:
> Only 2 decimals

OK 👌 I've also switched it: first AA, then frequency. @rtobes is it better?

![output](https://user-images.githubusercontent.com/766656/36913254-9d15707c-1e49-11e8-8ade-8da8581eb414.png)

@agarciamontoro I also removed `cut`, because if any of the chosen columns will have space, awk will still choke, so in the end I couldn't avoid reading awk docs... 😅 

```shell
awk -F'\t' '{ printf ">%.2f - %s\\n%s\\n", $4, $6, $5 }'
```

Check it out: `-F` for the separator, `printf` for the float rounding: `%.2f` 🤓

laughedelic commented on src/main/scala/loquats/phylogeneticTree.scala:
```diff
@@ -59,7 +59,7 @@ case object visualizations {
         val inputPath = productiveClonotypesTSV.getCanonicalPath()
         val outputPath = output(data.viz.phylogeneticTree).getCanonicalPath()
-        client eval s"""
+        client eval raw"""
```
@agarciamontoro I just remembered about `raw` interpolator, with it you don't need to double escape things:

```scala
@ raw"foo \n \\n ${1+1} $${1+1}"
res0: String = "foo \\n \\\\n 2 ${1+1}"
@ println(raw"foo \n \\n ${1+1} $${1+1}")
foo \n \\n 2 ${1+1}
```

agarciamontoro replied:
Oh, this is great, thanks for the tip!

agarciamontoro commented:
awk rocks!

agarciamontoro commented:
Why didn't we merge this? Any reason besides that I completely forgot (sorry)?

agarciamontoro commented:
> Any reason besides that I completely forgot (sorry)?

Nope. Merging.
